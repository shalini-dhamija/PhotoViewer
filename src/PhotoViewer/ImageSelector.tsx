
const brokenImages = [
    1, 24, 32, 36, 44, 47
];

function getImageUrls() {
    const urls = [];

    for (let i = 0; i < 50; i++) {
        if (!brokenImages.includes(i)) {
            const imageNumberString = i.toString().padStart(2, '0');
            urls.push(`https://picsum.photos/id/6${imageNumberString}/1600/900.jpg`)
        }
    }

    return urls;
}

export function ImageSelector(props:{setImageURL: React.Dispatch<React.SetStateAction<string>>}){
    const urls = getImageUrls();

    function handleClick(url: React.SetStateAction<string>){
        props.setImageURL(url);
      } 
    
    return (                
        <div>
            <h2>Select Your Photo</h2>
            <div className="gridContainer">
            {urls.map(url =>(
                <div className="gridItem"><img src={url} alt={url} className="thumbNailImage" onClick={() => handleClick(url)}/>
                </div>                
            ))}                    
            </div>
        </div>
        );
}
//export const imageUrls = getImageUrls();