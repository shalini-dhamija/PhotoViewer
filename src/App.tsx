import './App.css'
import {PhotoViewer} from './PhotoViewer/PhotoViewer.tsx'
import {ImageSelector} from './PhotoViewer/ImageSelector.tsx'
import { useState } from 'react';


export function App() {
  const [imageURL, setImageURL] = useState('');
     
  return (
    <div>      
        <PhotoViewer imageURL = {imageURL}/>
        <ImageSelector setImageURL = {setImageURL}/>
      </div>
  );
}
