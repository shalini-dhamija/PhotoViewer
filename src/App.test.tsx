
import { render } from '@testing-library/react';
import {App} from './App';
import {getImageUrls} from './PhotoViewer/ImageSelector.tsx';
//import renderer from 'react-test-renderer';
import * as renderer from 'react-test-renderer'

test('renders React Photo Viewer text', () => {
  const { getByText } = render(<App />);
  const textElement = getByText(/React Photo Viewer/i);
  expect(textElement).toBeInTheDocument();
});

describe('A unit test to check our imageUrl generation code', function () {    
    test('Broken URL should not be present', function () {
        const brokenURL = 'https://picsum.photos/id/601/1600/900.jpg';
        const urls = getImageUrls();
        urls.forEach( (url) => {expect(url).not.toBe(brokenURL)});        
    });
    
    test('check for the First Link', function () {
        const firstURL = 'https://picsum.photos/id/600/1600/900.jpg';
        const urls = getImageUrls();
        expect(urls[0]).toBe(firstURL);        
    });
});

describe('Regression / Snapshot test', function () {    
    test('Snapshot for App', async function () {
        const tree = renderer.create(<App />).toJSON();
        expect(tree).toMatchSnapshot();   
    });
});


