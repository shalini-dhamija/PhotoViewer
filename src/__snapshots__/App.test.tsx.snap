// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Regression / Snapshot test Snapshot for App 1`] = `
<div>
  <div>
    <h1>
      React Photo Viewer
    </h1>
    <div
      className="topImage"
    >
      <img
        alt="https://picsum.photos/id/600/1600/900.jpg"
        src="https://picsum.photos/id/600/1600/900.jpg"
      />
    </div>
  </div>
  <div>
    <h2>
      Select Your Photo
    </h2>
    <div
      className="gridContainer"
    >
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/600/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/600/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/602/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/602/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/603/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/603/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/604/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/604/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/605/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/605/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/606/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/606/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/607/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/607/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/608/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/608/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/609/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/609/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/610/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/610/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/611/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/611/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/612/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/612/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/613/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/613/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/614/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/614/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/615/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/615/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/616/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/616/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/617/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/617/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/618/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/618/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/619/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/619/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/620/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/620/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/621/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/621/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/622/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/622/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/623/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/623/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/625/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/625/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/626/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/626/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/627/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/627/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/628/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/628/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/629/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/629/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/630/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/630/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/631/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/631/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/633/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/633/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/634/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/634/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/635/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/635/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/637/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/637/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/638/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/638/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/639/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/639/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/640/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/640/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/641/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/641/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/642/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/642/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/643/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/643/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/645/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/645/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/646/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/646/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/648/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/648/1600/900.jpg"
        />
      </div>
      <div
        className="gridItem"
      >
        <img
          alt="https://picsum.photos/id/649/1600/900.jpg"
          className="thumbNailImage"
          onClick={[Function]}
          src="https://picsum.photos/id/649/1600/900.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;
